cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -826,6 +826,10 @@ void tcpProbe(String hostname, int port) throws IOException {
         }
     }
 
+    private String podDnsName(String podName) {
+        return DnsNameGenerator.podDnsName(namespace, KafkaResources.brokersServiceName(cluster), podName);
+    }
+
```

Comment:
Do we really need a method which just calls another method and is used from 2 places? TBH, I would say no.

(And if we would decide that this should stay ... maybe it can at least include the port as well then?)

Reply:
Given the root issue is inconsistent (and incorrect) addressing of pods a single method that gets it right seem appropriate to me. 

Including the port won't work for the TCP probe as the socket requires both address and port as separate arguments (either directly or via `InetSocketAddress`).

Reply:
There is a single consistent method ... its called `DnsNameGenerator.podDnsName`. It will not help with consistency if every class hides it into a separate layer of obfuscation.

Reply:
I've changed it to introduce an `Endpoint` record to pass back the host/port and make the new method do a little more work.